app/controllers/api/vms_controller.rb

```diff
@@ -477,5 +478,18 @@ def request_console_vm(vm, protocol)
     rescue => err
       action_result(false, err.to_s)
     end
+
+    def queue_options(method)
+      current_user = User.current_user
```

Comment:
Does current user always exists for API requests?

Reply:
yes, we set that based on the authentication (basic auth/auth/system).